exec/totemknet.c

```diff
@@ -1487,17 +1567,94 @@ int totemknet_reconfigure (
 	if (res) {
 		KNET_LOGSYS_PERROR(errno, LOGSYS_LEVEL_ERROR, "knet_handle_compress failed");
 	}
+	return res;
+}
+
+int totemknet_reconfigure (
+	void *knet_context,
+	struct totem_config *totem_config)
+{
+	struct totemknet_instance *instance = (struct totemknet_instance *)knet_context;
+	int res = 0;
+
+	(void)totemknet_configure_compression(knet_context, totem_config);
 
 #ifdef HAVE_LIBNOZZLE
-	/* Set up nozzle device(s). Return code is ignored, because unability
+	/* Set up nozzle device(s). Return code is ignored, because inability
 	 * configure nozzle is not fatal problem, errors are logged and
 	 * there is not much else we can do */
 	(void)setup_nozzle(instance);
 #endif
+
+	if (totem_config->crypto_changed) {
+		/* Flip crypto_index */
+		totem_config->crypto_index = 3-totem_config->crypto_index;
+		res = totemknet_set_knet_crypto(instance);
+
+		knet_log_printf(LOG_INFO, "kronosnet crypto reconfigured on index %d: %s/%s/%s", totem_config->crypto_index,
+				totem_config->crypto_model,
+				totem_config->crypto_cipher_type,
+				totem_config->crypto_hash_type);
+	}
 	return (res);
 }
 
 
+int totemknet_reconfigure_phase (
```

Comment:
Maybe consider change of name to something like crypto_reconfigure_phase, because right now, there is `totemknet_reconfigure` and `totemknet_reconfigure_phase`, so at the beginning I was wondering if `totemknet_reconfigure` is not just some shortcut to `totemknet_reconfigure_phase` with initial argument, but it is not.